libs/remix-ui/editor/src/lib/remix-ui-editor.tsx

```diff
@@ -541,6 +543,33 @@ export const EditorUI = (props: EditorUIProps) => {
       }
     })
 
+    editor.onDidPaste((e) => {
+       if (!pasteCodeRef.current && e && e.range && e.range.startLineNumber >= 0 && e.range.endLineNumber >= 0 && e.range.endLineNumber - e.range.startLineNumber > 10) {
+        const modalContent: AlertModal = {
+          id: 'newCodePasted',
+          title: 'Slow down, Cowboy! New pasted code detected.',
+          message: (
+            <div> <i className="fas fa-exclamation-triangle text-danger mr-1"></i>
+              You have just pasted a code snippet or contract in the editor.
+              <div>
+                Make sure you fully understand this new code before executing <b>any</b> transaction that uses it.
+                <div>
+                Your wallet might be exposed and <span className='text-warning'> at risk </span> if you run untrusted code. In a worst-case scenario, you could <span className='text-warning'>loose all your money</span>.
+                </div>
+                <span className='text-warning'>If you don't fully understand it, please don't run this code.</span>
+                <div>
+                If you are not a smart contract developer, please ask assistance to someone you trust and has the appropriate skills to determine whether this code is safe to use.
+                </div>
+                <div>see <a href='https://remix-ide.readthedocs.io/en/latest/security.html'> this blog post </a> for more information.</div>
```

Comment:
It says `this blog post` but actually that takes to a section of Remix docs. It opens the link in the same tab which closes Remix IDE.

`S` in `see` should be in capital letter.